### DISCUSSION TITLE
Should `uv_pipe_bind` support abstract socket UDS for Linux?

### DISCUSSION THREAD

**Discussion by cocktail828** [libuv/libuv#3815] [Ideas]
Recently, I use libuv to interprocess communication between containers.
As we all know UDS is much effective on local communication. However, currently `uv_pipe_bind` will generate an '.sock file' and the other container should also mount the '.sock file' too.
So I wonder whether libuv can support abstract socket UDS. There wont be too much modification for the feature. 
https://github.com/libuv/libuv/blob/2b4b293ebb2e876fe61df37c00d522ce55f2e34d/src/unix/pipe.c#L43

The following is an simple example.
```c
// There also another way to add the parameter `abstract`(add an member in `uv_pipe_t` for example).
int uv_pipe_bind(uv_pipe_t* handle, const char* name, bool abstract) {
  ...
  memset(&saddr, 0, sizeof saddr);
  if (abstract) {
      // set sun_path[0]='\0'
      uv__strscpy((char*)saddr.sun_path+1, pipe_fname, sizeof(saddr.sun_path));
  } else {
      uv__strscpy((char*)saddr.sun_path, pipe_fname, sizeof(saddr.sun_path));
  }
  saddr.sun_family = AF_UNIX;
  ...
```

**Comment by vtjnash** [comment]
Is that the same as https://github.com/libuv/libuv/pull/3096?

**Reply by cocktail828** [reply]
yep! Exactly the same.
